webui/react/src/hooks/useExperimentTags.ts

```diff
@@ -1,19 +1,26 @@
 import { useCallback } from 'react';
 
+import { TagAction, tagsActionHelper } from 'components/kit/Tags';
 import { patchExperiment } from 'services/api';
 
 export interface UseExperimentTagsInterface {
-  handleTagListChange: (experimentId: number) => (tags: string[]) => void;
+  handleTagListChange: (
+    experimentId: number,
+    tags: string[],
+  ) => (action: TagAction, tag: string, updatedId?: number) => void;
 }
 
 const useExperimentTags = (callbackFn?: () => void): UseExperimentTagsInterface => {
```

Comment:
separate from the changes to add `tagsActionHelper` here, I'm also questioning the usefulness of this hook in the first place. IMO the DRYness isn't worth it. would prefer the clarity of explicitly calling both `patchExperiment` and the refetch in the callback that's passed to the component.